File: M3 Intermediate DSA/src/L8_Arrays_2D_Matrices/Assignment2.java
package L8_Arrays_2D_Matrices;

public class Assignment2 {

  public int[] solve(int[][] A) {
    int N = A.length;
    int M = A[0].length;
    int[] ans = new int[N];

    for (int row = 0; row < N; row++) {
      int sum = 0;
      for (int col = 0; col < M; col++) {
        sum += A[row][col];
      }

      ans[row] = sum;
    }

    return ans;
  }

}

File: M3_Intermediate_DSA/L6_2D_Matrices/assignment2.go
package main

func SolveA2(A [][]int) []int {
	ans := make([]int, len(A))

	for rowIndex, row := range A {
		var sum int
		for _, col := range row {
			sum += col
		}

		ans[rowIndex] = sum
	}

	return ans
}
